extensions/data-workspace/src/dialogs/projectDashboard.ts

```diff
@@ -0,0 +1,124 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as constants from '../common/constants';
+import { IProjectAction, IProjectActionGroup, IProjectProvider, WorkspaceTreeItem } from 'dataworkspace';
+import { IWorkspaceService } from '../common/interfaces';
+
+export class ProjectDashboard {
+
+	private dashboard: azdata.window.ModelViewDashboard | undefined;
+	private modelView: azdata.ModelView | undefined;
+	private projectProvider: IProjectProvider | undefined;
+	private overviewTab: azdata.DashboardTab | undefined;
+
+	constructor(private workspaceService: IWorkspaceService, private treeItem: WorkspaceTreeItem) {
+	}
+
+	public async showDashboard(): Promise<void> {
+		const project = this.treeItem.element.project;
```

Comment:
can this ever be undefined?  might be a good idea to strongly-type `project` just in case.

Reply:
It should ideally not be undefined here. Since the element and project are indeed of type "any", I have added a couple of extra checks to make sure projectFilePath exists and projectFileName isn't null.